examples/user_guide/Network_Graphs.ipynb

```diff
@@ -0,0 +1,335 @@
+    "nodes = hv.Nodes((x, y, node_indices, ['Output']+['Input']*(N-1)),  vdims=['Type'])\n",
+    "hv.Graph(((source, target), nodes, paths)).redim.range(**padding)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Working with NetworkX"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NetworkX is a very useful library when working with network graphs and the Graph Element provides ways of importing a NetworkX Graph directly. Here we will load the Karate Club graph and use the ``circular_layout`` function provided by NetworkX to lay it out:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%opts Graph [tools=['hover']]\n",
+    "import networkx as nx\n",
+    "G = nx.karate_club_graph()\n",
+    "hv.Graph.from_networkx(G, nx.layout.circular_layout).redim.range(**padding)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Animating graphs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%opts Graph {+framewise}\n",
+    "layouts = [nx.layout.circular_layout, nx.layout.fruchterman_reingold_layout, nx.layout.spectral_layout,\n",
+    "           nx.layout.spring_layout, nx.layout.random_layout]\n",
+    "\n",
+    "G = nx.karate_club_graph()\n",
+    "hv.HoloMap({l.__name__: hv.Graph.from_networkx(G, l) for l in layouts},\n",
+    "           kdims=['Layout'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Real world graphs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As a final example let's look at a slightly larger graph. We will load a dataset of a Facebook network consisting a number of friendship groups identified by their ``'circle'``. We will load the edge and node data using pandas and then color each node by their friendship group using many of the things we learned above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%opts Nodes Graph [width=800 height=800 xaxis=None yaxis=None tools=['hover']]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%opts Graph [color_index='circle']\n",
+    "%%opts Graph (node_size=10 edge_line_width=1)\n",
+    "colors = ['#000000']+hv.Cycle('Category20').values\n",
+    "edges_df = pd.read_csv('../assets/fb_edges.csv')\n",
+    "fb_nodes = hv.Nodes(pd.read_csv('../assets/fb_nodes.csv')).sort()\n",
+    "fb_graph = hv.Graph((edges_df, fb_nodes), label='Facebook Circles')\n",
+    "fb_graph = fb_graph.redim.range(x=(-0.05, 1.05), y=(-0.05, 1.05)).opts(style=dict(cmap=colors))\n",
+    "fb_graph"
```

Comment:
Here the hover information seems like it should be "circle: 50", not "circle: circle50".  I know that means the original data needs updating, but I do think that's important to avoid being confusing.

Reply:
Or it could be something like "group: circle50"...

Reply:
Also, seems like the hover should have something (even if we have to make it up) to indicate the specific identity of the node.  I imagine these are anonymized, but presumably we need something like "identity: person2090" to show that we can use this information to learn not just what category it's in, but who it is (the node identity).

Reply:
If you pulled the latest version you should have the node index as well at least, is that sufficient?

Reply:
Ok, just pulled again and got the index info.  Yes, that's helpful, but in general people won't *only* want an index, they'll want some sort of human-meaningful information.  But that's presumably available if it's in a column, so we can just say that in words ("Hovering over the node will reveal the index of that node, along with any other columns of information about that node that are available")?

BTW, the index info isn't always coming out ok:

![image](https://user-images.githubusercontent.com/1695496/30495810-095470aa-9a13-11e7-80e0-91eb19425cb6.png)

Reply:
That's pretty damn weird.

Reply:
The ??? isn't the only odd bit, it's odd that it's trying to show *all* the node ids at all.  Maybe that's useful sometimes, but I'd guess probably not, since in general that can be a huge list and it's unclear what any of those other indexes would be (and unclear which is the one node I'm hovering over).  Or maybe those are edge indexes.  Confusing!

Reply:
>Or maybe those are edge indexes.

I suspect that's it, in which case I can definitely fix it.

Reply:
Fixed.